### DISCUSSION TITLE
Extrinsic Calibration

### DISCUSSION THREAD

**Discussion by dave992** [sam-xl/scancontrol#43] [General]
An extrinsic calibration is needed if the laser line scanner will be used together with a robot (or other moving platforms). 

So far I've only been able to find the following package to do a calibration of a laser line scanner (requires a camera): [laser_camera_calibration](http://wiki.ros.org/laser_camera_calibration)

Alternatively, we can implement another method that is not reliant on the use of a camera. This could, for example, be based on [A one-step intrinsic and extrinsic calibration method for laser line scanner operation in coordinate measuring machines](https://www.researchgate.net/publication/231150307_A_One-Step_Intrinsic_and_Extrinsic_Calibration_Method_for_Laser_Line_Scanner_Operation_in_Coordinate_Measuring_Machines)

Any suggestions on what route to take or packages I might be missing?

**Comment by gavanderhoorn** [comment]
Related: ros-industrial/industrial_calibration#95 and ros-industrial-consortium/godel#114.

I'm not sure whether @VictorLamoine ever made any of the code he is/was using available.

**Comment by gavanderhoorn** [comment]
Note: I don't believe this is really something 'core' to the driver. 2D laser calibration is something useful / needed regardless of which driver or even laser is being used.

**Comment by dave992** [comment]
> Related: ros-industrial/industrial_calibration#95 and ros-industrial-consortium/godel#114.

Thanks, I will have a look at it.

> Note: I don't believe this is really something 'core' to the driver. 2D laser calibration is something useful / needed regardless of which driver or even laser is being used.

Agreed, but for now it seemed the most related package for us to post this issue in. Will transfer or close it with a link once we find or create a laser line calibration package.

**Comment by gavanderhoorn** [comment]
Ping @VictorLamoine seeing as he just responded to a comment on ros-industrial/industrial_calibration#95 ;)

**Comment by VictorLamoine** [comment]
@gavanderhoorn I missed this one sorry :smile: 

Our procedure works but has never been put in a "production" ready state.

Interesting papers:
- https://hal.inria.fr/inria-00590039/document
- https://www.rob.uni-luebeck.de/index.php?id=276&author=0:2140&L=0

I will send you our Matlab code (can be used with Octave from what I remember) that worked with a PDF.
If you ever come to a working/automated solution please let me know!

**Comment by VictorLamoine** [comment]
@dave992 I don't have your email and I can't find it. I added you on LinkedIn so we can share emails!

**Comment by gavanderhoorn** [comment]
> @dave992 I don't have your email and I can't find it. I added you on LinkedIn so we can share emails!

You could send it to me @VictorLamoine. I can forward it to the relevant ppl (ie: @dave992).

**Comment by VictorLamoine** [comment]
You should both have received it!

**Comment by farshbafdoustar** [comment]
Can I have also the Matlab code? @VictorLamoine
Is there any progress here? @gavanderhoorn , @dave992 . I'm also interested in the topic and can contribute if there is any opportunity.
